File: modulewh/clientwh/src/main/resources/static/js/service/materialstoreService.js

/**
 * Service for Material Store
 **/

 define(['require', 'angular'], function (require, angular) {
 	app.aService(clientwh.prefix + 'materialstoreService', function ($http, $rootScope) {

		// Create.
		this.create = function (materialstore) {
			var serverUrl = clientwh.serverUrl + '/materialstore/create';
			var request = {
				method: 'POST',
				url: serverUrl,
				data: materialstore
			}
			return $http(request);
		}

		// Update Lock.
		this.updateLock = function(id) {
			var serverUrl = clientwh.serverUrl + '/materialstore/updateLock/' + id;
			var request = {
				method: 'PUT',
				url: serverUrl
			}
			return $http(request);
		}
		
		// Update Unlock.
		this.updateUnlock = function(id, materialstore) {
			var serverUrl = clientwh.serverUrl + '/materialstore/updateUnlock/' + id;
			var request = {
				method: 'PUT',
				url: serverUrl
			}
			return $http(request);
		}
		
		// Update With Lock.
		this.updateWithLock = function(id, materialstore) {
			var serverUrl = clientwh.serverUrl + '/materialstore/updateWithLock/' + id;
			var request = {
				method: 'PUT',
				url: serverUrl,
				data: materialstore
			}
			return $http(request);
		}
		
		// Update For Delete.
		this.updateForDelete = function(id) {
			var serverUrl = clientwh.serverUrl + '/materialstore/updateForDelete/' + id;
			var request = {
				method: 'PUT',
				url: serverUrl
			}
			return $http(request);
		}
		
		// Update For Delete With Lock.
		this.updateForDeleteWithLock = function(id) {
			var serverUrl = clientwh.serverUrl + '/materialstore/updateForDeleteWithLock/' + id;
			var request = {
				method: 'PUT',
				url: serverUrl
			}
			return $http(request);
		}
		
		// Update.
		this.update = function (id, materialstore) {
			var serverUrl = clientwh.serverUrl + '/materialstore/update/' + id;
			var request = {
				method: 'PUT',
				url: serverUrl,
				data: materialstore
			}
			return $http(request);
		}

		// Delete.
		this.delete = function (id) {
			var serverUrl = clientwh.serverUrl + '/materialstore/delete/' + id;
			var request = {
				method: 'DELETE',
				url: serverUrl
			}
			return $http(request);
		}

		// Get by Id.
		this.getById = function (id) {
			var serverUrl = clientwh.serverUrl + '/materialstore/getById/' + id;
			var request = {
				method: 'GET',
				url: serverUrl
			}
			return $http(request);
		}

		// List for page and filter.
		this.listWithCriteriasByPage = function (criterias, pageNo, pageSize, sorts) {
			var serverUrl = clientwh.serverUrl + '/materialstore/listWithCriteriasByPage?' + 'page=' + pageNo + '&size=' + pageSize;
			if (typeof (sorts) !== 'undefined' && sorts.length > 0) {
				angular.forEach(sorts, function (sort) {
					serverUrl += '&' + sort;
				});
			}
			var request = {
				method: 'POST',
				url: serverUrl,
				data: criterias
			}
			return $http(request);
		}
		// List for page and filter.
		this.listWithCriteriasByIdStoreAndPage = function(idstore, criterias, pageNo, pageSize) {
			var serverUrl = clientwh.serverUrl + '/materialstore/listWithCriteriasByIdStoreimportAndPage/' + idstore + '?page=' + pageNo + '&size=' + pageSize;
			var request = {
					 method: 'POST',
					 url: serverUrl,
					 data: criterias
					}
			return $http(request);
		}
		// List for page and filter.
		this.listWithCriteriasByIdMaterialAndPage = function(idmaterial, criterias, pageNo, pageSize) {
			var serverUrl = clientwh.serverUrl + '/materialstore/listWithCriteriasByIdMaterialimportAndPage/' + idmaterial + '?page=' + pageNo + '&size=' + pageSize;
			var request = {
					 method: 'POST',
					 url: serverUrl,
					 data: criterias
					}
			return $http(request);
		}
		
		// Update With Lock.
		this.checkQuantityMaterial = function(idstore, materialstore) {
			var serverUrl = clientwh.serverUrl + '/materialstore/checkQuantityMaterial/' + idstore;
			var request = {
				method: 'POST',
				url: serverUrl,
				data: materialstore
			}
			return $http(request);
		}
	});
 });

File: modulewh/serverwh/src/main/java/com/redsun/server/wh/service/MaterialstoreService.java

package com.redsun.server.wh.service;

import java.io.IOException;
import java.util.List;
import java.util.Map;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.redsun.server.wh.model.Materialstore;
import com.redsun.server.wh.model.common.SearchCriteria;

public interface MaterialstoreService {

	/**
	 * Saves (create or update) the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialstore
	 * @return
	 */
	Materialstore save(Materialstore materialstore);

	/**
	 * Create the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialstore
	 * @return
	 */
	Materialstore create(Materialstore materialstore) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Integer updateLock(Integer id) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @param idlock
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Integer updateUnlock(Integer id) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialstore
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialstore updateWithLock(Integer id, Materialstore materialstore) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialstore
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialstore updateForDelete(Integer id, Integer version) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialstore
	 * @return
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	Materialstore updateForDeleteWithLock(Integer id, Integer version) throws JsonParseException, JsonMappingException, IOException;
	
	/**
	 * Update the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialstore
	 * @return
	 */
	Materialstore update(Integer id, Materialstore materialstore) throws JsonParseException, JsonMappingException, IOException;

	/**
	 * Deletes the given entity <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param materialstore
	 * @return true if found and deleted, false if not found
	 */
	void delete(Materialstore materialstore);

	/**
	 * Deletes the entity by its Primary Key <br>
	 * Transactional operation ( begin transaction and commit )
	 * @param id
	 * @return true if found and deleted, false if not found
	 */
	void deleteById(Integer id);

	/**
	 * Loads the entity for the given Primary Key <br>
	 * @param id
	 * @return the entity loaded (or null if not found)
	 */
	Materialstore getById(Integer id);

	/**
	 * Loads ALL the entities (use with caution)
	 * @return
	 */
	List<Materialstore> listAll();

	/**
	 * Count all the occurrences
	 * @return
	 */
	long countAll();

	/**
	 * Check exist
	 * @return
	 */
	boolean isExist(Integer id);

	/**
	 * Search by a criteria.
	 * @return
	 */
	List<Materialstore> listWithCritera(SearchCriteria searchCriteria);

	/**
	 * Search by multiple criteria.
	 * @return
	 */
	List<Materialstore> listWithCriteras(List<SearchCriteria> searchCriterias);

	/**
	 * Loads ALL the entities by page (use with caution)
	 * @return
	 */
	Page<Materialstore> listAllByPage(Pageable pageable);

	/**
	 * Search by a criteria by page.
	 * @return
	 */
	Page<Materialstore> listWithCriteraByPage(SearchCriteria searchCriteria, Pageable pageable);

	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<Materialstore> listWithCriterasByPage(List<SearchCriteria> searchCriterias, Pageable pageable);
	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Map<String, Object> quantityMaterialByidStore(Integer idstore,Integer idmaterial);
	
	Integer storeQuantity(Integer idstore);

	Materialstore updateWithLockForImport(Integer id, Materialstore materialstore)
			throws JsonParseException, JsonMappingException, IOException;
	
	/**
	 * Search by multiple criteria by page.
	 * @return
	 */
	Page<Map<String, Object>> listWithCriteriasByIdStoreAndPage(Integer idstore, List<SearchCriteria> searchCriterias,
			Pageable pageable);
	Page<Map<String, Object>> listWithCriteriasByIdMaterialAndPage(Integer idmaterial, List<SearchCriteria> searchCriterias,
			Pageable pageable);
	
	Map<String, Object> checkQuantityMaterial(Integer idstore,Integer idmaterial);
}
